ntbs-service/Pages/Index.cshtml.cs

```diff
@@ -41,6 +43,7 @@ public class IndexModel : PageModel
         
         public async Task OnGetAsync()
         {
+            BreadcrumbsHelper.SetTopLevelBreadcrumb(HttpContext.Session, "Index", "/Index");
```

Comment:
This is a great improvement on the interface - really nice!
I wander if at this stage you just want to make this an extension method on the `Session`'s type?